python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -275,24 +276,22 @@ def all_inputs_done(self):
         # once the bundle queue is exhausted.
         self._inputs_done = True
 
+        if self._metrics.num_inputs_received < self._actor_pool.min_size():
+            warnings.warn(
+                f"The minimum number of concurrent actors for '{self.name}' is set to "
+                f"{self._actor_pool.min_size()}, but the operator only received "
+                f"{self._metrics.num_inputs_received} input(s). This means that the "
+                f"operator can launch at most {self._metrics.num_inputs_received} "
+                f"task(s), and won't fully utilize the available concurrency. You "
+                "might be able to increase the number of concurrent tasks by "
+                "configuring `override_num_blocks` earlier in the pipeline."
+            )
+
```

Comment:
Unlike the other (removed) warning, Ray Data will emit this even if the pipeline doesn't finish.